backend/src/apiserver/resource/resource_manager.go

```diff
@@ -229,6 +233,56 @@ func (r *ResourceManager) GetPipelineTemplate(pipelineId string) ([]byte, error)
 	return template, nil
 }
 
+// user-gcp-sa secret was the default way to authenticate in GCP, but it is no longer needed when
+// we suggest using default authentication methods provided by GKE.
+// Alternatives include: project default service account, node pool default service account and
+// workload identity.
+// Dropping GOOGLE_APPLICATION_CREDENTIALS env when the secret is not configured ensures old
+// pipelines can still be run in new recommended clusters set up as above methods.
+//
+// TODO: when all pipeline samples no longer use user-gcp-sa secret, we can remove this usage.
+func (r *ResourceManager) dropUserGcpSaIfNotConfigured(workflow util.Workflow) {
+	someStepsHaveUserGcpSa := false
+	for _, template := range workflow.Workflow.Spec.Templates {
```

Comment:
For older pipelines you might also want to check `for _, volume := range workflow.Workflow.Spec.Volumes {`

Reply:
Thanks for catching this!

I will add it.

Reply:
Added